### PR TITLE
Knollfear/3262 auth affiliations audit table

### LINKED ISSUE #3262: [Feature] Create audit table for affiliations and create an expiration date for affiliations
**Description and related issues -**
1. Create a table called auth_affiliation_audit, it should store affiliation_id, change_date, changed_by (oktaUser), new_role_id, original_role_id, original_status and new_status
3. Update the /states/:stateId/affiliations/:id end point so that when a user is approved/revoked/denied, it adds a row to the auth_affiliations_audit table to record the change.

### This task is done when…

- [ ] there is an audit table that records every time an affiliation is changed
- [ ] the affiliation table no longer has updated_at or updated_by

### PR DESCRIPTION
**Pull request by knollfear**
resolves #3262

**Description-**
Adds an audit table to capture changes to the auth affiliations table.  

**This pull request changes...**

- the process that occurs when a user's auth_affiliation status is changed.
- the npm commands down and rollback have been renamed to migrate:down and migrate:rollback respectively

**This pull request also touches…**

The changing of auth affiliations

- possible side effects of this change
The whole user approval/revoke process should be fully tested as it was refactored.

**This pull request was tested in the follow ways…**

- many users were approved and denied access to their state

**Steps to manually verify this change...**

1. Run the migration process as this requires a DB change (migrate:down should work too)
2. log in as a state admin
3. approve or revoke auth_affiliation requests as you see fit
4. check the DB with ```select * from auth_affiliation_audit;``` to make sure rows are populated as expected

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by tbolt — APPROVED** [review]
Code looks good. Tested it out and it was successfully saving auth affiliation changes to the database. I think we will need to sync up and sort through how best to merge this + https://github.com/CMSgov/eAPD/pull/3360/files as there could be some merge pain as-is.

File: api/migrations/20210804195502_auth_affiliation_audit.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    table
+      .string('new_status')
+
+  })
+
+  await knex.schema.table('auth_affiliations', table => {
+    table.dropColumn('updated_by')
+    table
+      .dateTime('expires_at')
```
[the remark this answers is not in the document]
Comment (knollfear) [reply api/migrations/20210804195502_auth_affiliation_audit.js:31]: let's go with date.  The hour of the day shouldn't matter.  I don't have a good justification for datetime over date.

**Comment by thetif** [comment]
@knoll I changed the migration to remove expires_at because Ty's PR is adding it. Also, we need to be storing the user_id for the user that is being updated.

**Comment by thetif** [comment]
Tested by:

|Given|When|Then|Result|
|---|---|---|---|
|Logged in as state admin|approves an affiliation request|adds the correct entry in the affiliations_audit table|:white_check_mark: PASS|
|Logged in as state admin|changes the role of an affiliation|adds the correct entry in the affiliation_audit table|:white_check_mark: PASS|
|Logged in as state admin|revoke the role of an affiliation|adds the correct entry in affiliation_audit table|:white_check_mark: PASS|
|Logged in as state admin|restore the role of an affiliation|adds the correct entry in the affiliation_audit table|:white_check_mark: PASS|
|Logged in as state admin|denies and affiliation request|adds the correct entry in the affiliation_audit table|:white_check_mark: PASS|
|Logged in as fed admin|approves an affiliation request|adds the correct entry in the affiliations_audit table|:white_check_mark: PASS|
|Logged in as fed admin|revoke the role of an affiliation|adds the correct entry in affiliation_audit table|:white_check_mark: PASS|
|Logged in as fed admin|restore the role of an affiliation|adds the correct entry in the affiliation_audit table|:white_check_mark: PASS|
|Logged in as fed admin|denies and affiliation request|adds the correct entry in the affiliation_audit table|:white_check_mark: PASS|

The existing functionality of approving, changing, revoking, denying, and restoring still works as expected.

**Comment by thetif** [comment]
@jeromeleecms you can test that everything is still functioning as you expect it to, but you won't be able to see the database unless you run it locally.

**Comment by thetif** [comment]
@knollfear my only question is should seed empty the affiliations_audit table when it empties the affiliations table?

**Comment by jeromeleecms** [comment]
Should I be seeing non state admins on the "Active" panel?

**Review by jeromeleecms — APPROVED** [review]
Talked to @thetif this morning about what the views and how this will transition to the all users tab in the new fed admin panel. This is good to go on my end.

**Pull request merged**